File: src/main/java/com/lucas/matches/infrastructure/document/SummaryType.java
package com.lucas.matches.infrastructure.document;

import java.util.HashMap;
import java.util.Map;

public enum SummaryType {
    AVB("AvB"),
    AVBTIME("AvBTime");

    private final String summaryTypeText;
    private static final Map<String, SummaryType> map = new HashMap<>();
    SummaryType(String summaryTypeText) {
        this.summaryTypeText = summaryTypeText;
    }
    static {
        for (SummaryType summaryType : SummaryType.values()) {
            map.put(summaryType.summaryTypeText, summaryType);
        }
    }

    public static SummaryType value(String summaryTypeText) {
        return map.get(summaryTypeText);
    }
}

File: src/main/kotlin/com/lucas/matches/service/domain/SummaryType.kt
package com.lucas.matches.service.domain

enum class SummaryType(private val summaryTypeText: String) {
    AVB("AvB"), AVBTIME("AvBTime");

    companion object {
        private val map: MutableMap<String, SummaryType> = HashMap()
        fun value(summaryTypeText: String): SummaryType? {
            return map[summaryTypeText]
        }

        init {
            for (summaryType in values()) {
                map[summaryType.summaryTypeText] =
                    summaryType
            }
        }
    }
}